roles/kubernetes/master/tasks/kube-proxy.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Lay down kube-proxy manifests
+  action: "{{ item.module }}"
+  args:
+    src: "{{ item.file }}{% if item.module == 'template' %}.j2{% endif %}"
+    dest: "{{ kube_config_dir }}/{{ item.file }}"
+  run_once: yes
+  register: kube_proxy_manifests
+  with_items:
+    - { name: kube-proxy, module: template, file: kube-proxy-ds.yaml, type: daemonset }
+    - { name: kube-proxy, module: template, file: kube-proxy-cm.yaml, type: configmap }
+
+- name: Start kube-proxy
+  kube:
+    name: "{{ item.item.name }}"
+    namespace: "kube-system"
+    kubectl: "{{ bin_dir }}/kubectl"
+    resource: "{{ item.item.type }}"
+    filename: "{{ kube_config_dir }}/{{ item.item.file }}"
+    state: "{{ kube_proxy_remove | ternary('absent', 'latest') }}"
```

Comment:
Does this work? If the file exists in the manifest folder, dosen't kubelet autostart it on reboots?

Is it better to .remove the entire file instead in the previous task ?

Reply:
It's not in manifest folder. It's a daemonset that has to be started with kubectl apply. Deleting the file is well, ideal, but this logic seems nice.